### ISSUE TITLE
Websocket regression in 0.9.98

### ISSUE THREAD

**Issue by themadsens** [Hammerspoon/hammerspoon#3422]
My websocket integration with Amplex GridLight broke in 0.9.98

Immediately after opening the **url: wss://demo.gridlight.dk/changes**

I get a callback with **_type="fail", message=nil_**

with 0.9.97 I get (as expected) **_type="open", message=nil_** and then **_type="received", message="authenticate"_**

**Comment by cmsj** [comment]
Thanks for the report. I upgraded the websocket library we use, which may be related. Is there a simple snippet I can use to reproduce what you're seeing? (Or would that need authentication?)

**Comment by themadsens** [comment]
Thank for noticing :-). No, the following snippet should be enough to reproduce the above behaviour:
The server is online, and the error actually prevents the "authenticate" step. It does not even get to saying "open"

```
  local wsUrl = "wss://devel.gridlight.dk/changes"
  print("WS open", wsUrl)
  ws = hs.websocket.new(wsUrl, function(typ, msg) 
      print("WS:", typ, msg)
  end)
```

**Comment by cmsj** [comment]
```lua
ws = hs.websocket.new("wss://demo.gridlight.dk/changes", function(state, message) print(state) print(message) end)
```
appears to be sufficient to reproduce what you're seeing.

**Comment by cmsj** [comment]
Yeah this is definitely broken. Looks like it might be an issue in SocketRocket (the underlying library we're using for websockets), which, worryingly, hasn't been updated since 2021. I'll revert to the earlier version of the library (PR is up already, #3424 ) and get a new Hammerspoon release out. Thanks again!

**Comment by Moulick** [comment]
https://github.com/Homebrew/homebrew-cask/pull/142987

**Comment by latenitefilms** [comment]
I wonder if this change has anything to do with it?

https://github.com/Hammerspoon/hammerspoon/commit/527194325391c322b7876be248b8d1a8af7662f4

**Comment by cmsj** [comment]
@latenitefilms I don't think so, that was just the underlying library separating out one method into two, but I dunno, better to revert quickly and take our time in figuring out a way forward (which may involve moving away from SocketRocket entirely - NSURLSession supports websockets in our minimum macOS version)
